csharp/src/Ice/Internal/TraceUtil.cs

```diff
@@ -411,11 +411,11 @@ private static byte printMessage(System.IO.StringWriter s, Ice.InputStream str,
 
         if (connection is not null)
         {
-            var connectionInfo = connection.getInfo();
```

Comment:
In C++, getting the connection info results in a self deadlock. I updated the code everywhere to get the connection ID through another path.

Reply:
Maybe d add a comment in the C++ side about the self deadlock.

Reply:
Done.